trio/_dtls.py

```diff
@@ -901,17 +958,19 @@ async def do_handshake(self, *, initial_retransmit_timeout=1.0):
                 return
 
             timeout = initial_retransmit_timeout
-            volley_messages = []
+            volley_messages: list[_AnyHandshakeMessage] = []
             volley_failed_sends = 0
 
-            def read_volley():
+            def read_volley() -> list[_AnyHandshakeMessage]:
                 volley_bytes = _read_loop(self._ssl.bio_read)
                 new_volley_messages = decode_volley_trusted(volley_bytes)
                 if (
                     new_volley_messages
                     and volley_messages
                     and isinstance(new_volley_messages[0], HandshakeMessage)
-                    and new_volley_messages[0].msg_seq == volley_messages[0].msg_seq
+                    # TODO: add isinstance or do a cast?
+                    and new_volley_messages[0].msg_seq
+                    == cast(HandshakeMessage, volley_messages[0]).msg_seq
```

Comment:
I looked at this a couple minutes and I don't really understand why we assume `volley_messages[0]` is a `HandshakeMessage`. This might be some sort of invariant in DTLS so IDK... Probably no harm in adding an `isinstance`?

Reply:
It seems to be a `HandshakeMessage` in the tests, and the only other datatype (in that file) with `msg_seq` is `HandshakeFragment`, so I suppose it should be fine.